src/views/ProfileCreation/UserName.tsx

```diff
@@ -81,28 +83,56 @@ const UserName: React.FC = () => {
   )
   const isUserCreated = existingUserState === ExistingUserState.CREATED
 
-  const checkUsernameValidity = debounce(async (value: string) => {
-    try {
-      setIsLoading(true)
-      const res = await fetch(`${API_PROFILE}/api/users/valid/${value}`)
+  const [usernameAvailability, setUsernameAvailability] = useState<string>(undefined)
+  const debouncedUsernameAvailability = useDebounce(usernameAvailability, 200)
 
-      if (res.ok) {
-        setIsValid(true)
-        setMessage('')
-      } else {
-        const data = await res.json()
+  useEffect(() => {
+    const fetchUsernameAvailability = async (abortSignal) => {
+      try {
+        setIsLoading(true)
+        if (!debouncedUsernameAvailability) {
+          setIsValid(false)
+          setMessage('')
+          fetchAbortSignal.current = null
+        } else {
+          const res = (await fetchWithTimeout(`${API_PROFILE}/api/users/valid/${debouncedUsernameAvailability}`, {
+            method: 'get',
+            signal: abortSignal,
+            timeout: 30000,
+          })) as Response
```

Comment:
Response type can be typed in `fetchWithTimeout` as `Promise<Response>`.

Reply:
Done